src/app/features/tech-record/components/test-record-summary/test-record-summary.component.ts

```diff
@@ -30,15 +40,38 @@ export class TestRecordSummaryComponent {
     return testResult.testTypes.map(t => t.testResult).join(',');
   }
 
+  get sortedTestTypeFields(): TestField[] {
+    const arr: TestField[] = [];
+
+    this.testRecords.forEach(record =>
+      record.testTypes.forEach(t => {
+        const field: TestField = {
+          testTypeStartTimestamp: t.testTypeStartTimestamp,
+          testTypeName: t.testTypeName,
+          testNumber: t.testNumber,
+          testResult: t.testResult,
+          testResultId: record.testResultId
+        };
+        arr.push(field);
+      })
+    );
+
+    return arr.sort((a, b) => new Date(b.testTypeStartTimestamp).getTime() - new Date(a.testTypeStartTimestamp).getTime());
```

Comment:
As you're just converting one type of array into a different type of array, using `map()` might be more relevant to what you want to achieve.

```suggestion
    const byDate = (a: TestField, b: TestField) => new Date(b.testTypeStartTimestamp).getTime() - new Date(a.testTypeStartTimestamp).getTime();

    return this.testRecords
      .map(record => record.testTypes
        .map(testType => ({
            testTypeStartTimestamp: testType.testTypeStartTimestamp,
            testTypeName: testType.testTypeName,
            testNumber: testType.testNumber,
            testResult: testType.testResult,
            testResultId: record.testResultId
          })
        )
      )
      .flat()
      .sort(byDate);
```

Reply:
`flatMap` is even better 🚀

Reply:
(modified my initial suggestion to use `flatMap()` instead)